### PR TITLE
Allow LDAP Sources to provide Avatars

### LINKED ISSUE #4144: New Field Request: Ability to set Profile Pic with LDAP
It would be neat to be able to set profile picture with LDAP. I suppose this would be best implemented as a URL, but I know there are multiple ways of implementing images in ldap.

### PR DESCRIPTION
**Pull request by zeripath**
Add setting to LDAP source to allow it to provide an Avatar.

Currently this is required to point to the image bytes.

Fix #4144

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by silverwind** [comment]
I assume it has to be JPEG for legacy reasons and can not support newer formats like PNG?

**Comment by zeripath** [comment]
Gitea doesn't care but I LDAP requires JPEG.

**Comment by zeripath** [comment]
I have been able to test with the gitea/test-openldap and this now works.

File: services/auth/source/ldap/source_sync.go [hunk 1]
```diff
@@ -112,12 +112,18 @@ func (source *Source) Sync(ctx context.Context, updateExisting bool) error {
...
+			}
+
+			if err == nil && isAttributeSSHPublicKeySet {
 				log.Trace("SyncExternalUsers[%s]: Adding LDAP Public SSH Keys for user %s", source.loginSource.Name, usr.Name)
 				if models.AddPublicKeysBySource(usr, source.loginSource, su.SSHPublicKey) {
 					sshKeysNeedUpdate = true
 				}
 			}
+
+			if err == nil && source.AttributeAvatar != "" {
```
Comment (lafriks) [inline services/auth/source/ldap/source_sync.go]:
```suggestion
			if err == nil && len(source.AttributeAvatar) != 0 {
```

File: services/auth/source/ldap/source_sync.go [hunk 2]
```diff
@@ -150,6 +156,13 @@ func (source *Source) Sync(ctx context.Context, updateExisting bool) error {
 					log.Error("SyncExternalUsers[%s]: Error updating user %s: %v", source.loginSource.Name, usr.Name, err)
 				}
 			}
+
+			if usr.IsUploadAvatarChanged(su.Avatar) {
+				if err == nil && source.AttributeAvatar != "" {
```
Comment (lafriks) [inline services/auth/source/ldap/source_sync.go]:
```suggestion
				if err == nil && len(source.AttributeAvatar) != 0 {
```

**Review by lafriks — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**